File: Strings/1024/1024.java
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);

		int qtdTestes = Integer.parseInt(scan.nextLine());

		while (qtdTestes-- > 0) {
			StringBuilder texto = new StringBuilder(scan.nextLine());

			primeiraPassada(texto);
			segundaPassada(texto);
			terceiraPassada(texto);
			
			System.out.println(texto);
		}
	}

	private static void primeiraPassada(StringBuilder texto) {
		for (int i = 0; i < texto.length(); i++) {
			char caractere = texto.charAt(i);

			if (caractere >= 65 && caractere <= 90 || caractere >= 97 && caractere <= 122) {
				caractere += 3;
				texto.setCharAt(i, caractere);
			}
		}
	}

	private static String segundaPassada(StringBuilder texto) {
		return texto.reverse().toString();
	}

	private static void terceiraPassada(StringBuilder texto) {
		for (int i = texto.length() / 2; i < texto.length(); i++) {
			char caractere = texto.charAt(i);
			caractere -= 1;
			texto.setCharAt(i, caractere);
		}
	}
}

File: Strings/1024/1024.cpp
#include <bits/stdc++.h>
#include <iostream>

using namespace std;

void primeiraPassada(string *, int);
void segundaPassada(string *);
void terceiraPassada(string *, int);

int main ()
{
	int qtd;
	string teste;
	cin >> qtd;
	getline (cin,teste);
	
	while(qtd--){
		string texto;

		getline (cin,texto);
		int size = texto.length();
		
		primeiraPassada(&texto, size);
		segundaPassada(&texto);
		terceiraPassada(&texto, size);

		cout << texto << endl;	
	}    

  return 0;
}

void primeiraPassada(string * texto, int size) {
    for(int i=0; i<size; i++) {
        int caractere = (*texto)[i];
        if ((caractere >= 65 && caractere <= 90) || (caractere >= 97 && caractere <= 122)) {
			(*texto)[i] += 3;
		}
    }
}

void segundaPassada(string * texto) {
    reverse((*texto).begin(), (*texto).end());
}

void terceiraPassada(string * texto, int size) {
    for(int i=size/2; i<size; i++) {
        (*texto)[i] -= 1;
    }
}